### PR TITLE
Nixify cardano-db-sync

### PR DESCRIPTION
**Pull request by craigem**
ToDo:

- [x] bors
- [ ] test the initial commit
- [ ] test containers

### CODE REVIEW

**Comment by craigem** [comment]
bors try

**Comment by disassembler** [comment]
bors ping

**Comment by disassembler** [comment]
bors try

**Comment by erikd** [comment]
Would be nice to squash this PR down a bit.

**Comment by cleverca22** [comment]
bors try

**Comment by cleverca22** [comment]
bors try

File: nix/docker.nix [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    created = "now";   # Set creation date to build time. Breaks reproducibility
+    contents = [ entry-point ];
+    config = {
+      EntryPoint = [ "${entry-point}/bin/entry-point" ];
+    };
+  };
+
+in {
+  dbSync = dbSyncDockerImage;
+  #dbSyncExtended = dbSyncExtendedDockerImage;
```
Comment (rhyslbw) [inline nix/docker.nix]: This service also requires an image. It was commented out for clarity while the standard version was being implemented. It has the same interface and requirements

File: nix/sources.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{
+    "cardano-node": {
```
Comment (rhyslbw) [inline nix/sources.json]: Does this need to be a dependency? There's currently no integration test, and I can't see it being used

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+  - label: 'check-cabal-project'
+    command: 'nix-build ./nix -A iohkNix.checkCabalProject -o check-cabal-project.sh && ./check-cabal-project.sh'
+    agents:
+      system: x86_64-linux
+
+  - label: 'stack-cabal-sync'
+    command: 'nix-shell ./nix -A iohkNix.stack-cabal-sync-shell --run .buildkite/stack-cabal-sync.sh'
+    agents:
+      system: x86_64-linux
```
Comment (rhyslbw) [inline .buildkite/pipeline.yml]: Please also add the build and push job, as per the reference: https://github.com/input-output-hk/cardano-node/blob/master/.buildkite/pipeline.yml#L41-L48

File: nix/docker.nix [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  dbSyncDockerImage = let
+    clusterStatements = lib.concatStringsSep "\n" (lib.mapAttrsToList (_: value: value) (commonLib.forEnvironments (env: ''
+      elif [[ "$NETWORK" == "${env.name}" ]]; then
+        ${if (env ? nodeConfig) then "exec ${scripts.${env.name}.db-sync}"
+        else "echo db-sync not supported on ${env.name} ; exit 1"}
+    '')));
+    entry-point = writeScriptBin "entry-point" ''
+      #!${runtimeShell}
+      # set up /tmp (override with TMPDIR variable)
+      mkdir -m 1777 tmp
```
Comment (rhyslbw) [inline nix/docker.nix]: The [schema dir](https://github.com/input-output-hk/cardano-db-sync/tree/master/schema) needs to be copied into the container at /schema

File: nix/docker.nix [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+      if [[ -d /config ]]; then
```
Comment (rhyslbw) [inline nix/docker.nix]: Need to set the ENV  here `export PGPASSFILE=/config/pgpass`

File: nix/docker.nix (hunk 4 above)
Comment (cleverca22) [reply nix/docker.nix]: oh, i fixed that in `nix/nixos/cardano-db-sync-service.nix` but didnt notice it in `nix/docker.nix`
i'll fix it with `${` and you can comment on if thats enough or not

**Pull request merged**